src/translib/transformer/xfmr_vrf.go

```diff
@@ -231,28 +243,69 @@ var network_instance_table_name_xfmr TableXfmrFunc = func (inParams XfmrParams)
                 return tblList, errors.New("Invalide network instance name")
         }
 
-        /*
-         * for GET or DELETE, use the key "name" only to get the DB table name
-         * for CREATE, UPDATE or REPLACE, use the key "name" and type to get the DB table name
-         */
+        /* For CREATE or PATCH at top level (Network_instances), check the config type if user provides one */
         if ((inParams.oper == CREATE) ||
-            (inParams.oper == UPDATE) ||
-            (inParams.oper == REPLACE))  {
-                /* get the type for the network instance config */
+            ((inParams.oper == UPDATE) && (inParams.requestUri == "/openconfig-network-instance:network-instances"))) {
                 oc_nwInstType, ierr := getNwInstType(nwInstObj, keyName)
                 if (ierr != nil ) {
-                        log.Info("network_instance_table_name_xfmr, network instance type not correct")
+                        log.Info("network_instance_table_name_xfmr, network instance type not correct ", oc_nwInstType)
                         return tblList, errors.New("network instance type incorrect")
                 }
 
                 log.Info("network_instance_table_name_xfmr, name ", keyName)
                 log.Info("network_instance_table_name_xfmr, type ", oc_nwInstType)
 
                 tblList = append(tblList, NwInstTblNameMapWithNameAndType[NwInstMapKey{intNwInstName, oc_nwInstType}])
```

Comment:
Is this append required? or just assignment is good enough?

Reply:
Just assign won't work as the return from NwInstTblNameMapxxx is a string, and tblList is []string. This is used by other xfmr table name functions as well.